backend/app/notify/email.go

```diff
@@ -0,0 +1,242 @@
+package notify
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net"
+	"net/smtp"
+	"sync"
+	"text/template"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+	"github.com/go-pkgz/repeater"
+	"github.com/pkg/errors"
+)
+
+// EmailParams contain settings for email set up
+type EmailParams struct {
+	Host     string        // SMTP host
+	Port     int           // SMTP port
+	TLS      bool          // TLS auth
+	From     string        // From email field
+	Username string        // user name
+	Password string        // password
+	TimeOut  time.Duration // TLS connection timeout
+	Template string        // request message template
+}
+
+// Email implements notify.Destination for email
+type Email struct {
+	EmailParams
+	SMTPClient
+	template  *template.Template // request message template
+	sendMutex sync.Mutex         // Send is synchronious and blocked by this mutex
+	count     int                // amount of waiting + running send requests
+}
+
+// SMTPClient interface defines subset of net/smtp used by email client
+type SMTPClient interface {
+	Mail(string) error
+	Auth(smtp.Auth) error
+	Rcpt(string) error
+	Data() (io.WriteCloser, error)
+	Quit() error
+	Close() error
+}
+
+// tmplData store data for message from request template execution
+type tmplData struct {
+	From      string
+	To        string
+	Orig      string
+	Link      string
+	PostTitle string
+}
+
+const emailConnectionTimeOut = 10 * time.Second
```

Comment:
calling it defaultEmailConnectionTimeout or just defaultEmailTimeout will clarify the nature of the const